core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -529,20 +564,33 @@ class BrokerTopicMetrics(name: Option[String], configOpt: java.util.Optional[Kaf
   }
 }
 
-class BrokerTopicAggregatedMetric() {
+class BrokerTopicAggregatedMetric(allTopics: Boolean = false) {
   private val partitionMetricValues = new ConcurrentHashMap[Int, Long]()
+  private val topicMetricValues = new ConcurrentHashMap[String, Long]()
 
   def setPartitionMetricValue(partition: Int, partitionValue: Long): Unit = {
     partitionMetricValues.put(partition, partitionValue)
   }
 
+  def setTopicMetricValue(topic: String, topicValue: Long): Unit = {
```

Comment:
do we want to remove the topic metric after the topic is deleted?

Reply:
Ah, I know what you mean now. Give me some time to fix it. Thanks for pointing it out!